tools/AzureRM.Example.psm1

```
@@ -9,6 +9,51 @@
 $PSDefaultParameterValues.Clear()
 Set-StrictMode -Version Latest
 
+function Test-DotNet
+{
+    try
+    {
+        if (Get-ChildItem 'HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\' -ErrorAction Stop | Get-ItemPropertyValue -Name Release | Foreach-Object { $_ -lt 461808 })
+        {
+            throw ".NET Framework versions lower than 4.7.2 are not supported in Az.  Please upgrade to .NET Framework 4.7.2 or higher."
+        }
+    }
+    catch
+    {
+        Write-Verbose ".NET Framework version check failed."
+    }
+}
+
+if (%ISAZMODULE%)
+{
+    if ($PSVersionTable.PSVersion -lt [Version]'5.1')
+    {
+        throw "PowerShell versions lower than 5.1 are not supported in Az. Please upgrade to PowerShell 5.1 or higher."
+    }
+
+    Test-DotNet
+}
+
+try
+{
+    if (Test-Path -Path "$PSScriptRoot\StartupScripts")
+    {
+        Get-ChildItem "$PSScriptRoot\StartupScripts" | ForEach-Object {
+            . $_.FullName
+        }
+    }
+}
+catch
+{
+    throw "Import-Module failed."
```

I don't think we want to catch here, I think we want to surface the error